ddtrace/span.py

```diff
@@ -101,13 +97,40 @@ def __init__(
         self.parent_id = parent_id
         self.tracer = tracer
 
-        # sampling
-        self.sampled = True
-
-        self._context = context
         self._parent = None
         self._ignored_exceptions = None  # type: Optional[List[Exception]]
 
+    @property
+    def context(self):
+        # type: () -> Context
+        from .tracer import _get_trace
+
+        trace = _get_trace(self.trace_id)
```

Comment:
I can't put my finger on it, but something feels off to me here...

Reply:
lol if we put `_get_trace` on `Tracer` then you'd probably be more at ease with

```suggestion
        trace = self.tracer._get_trace(self.trace_id)
```

Although this API exists only for backwards compatibility. I'd be in favour of removing it in the near future.

Reply:
this is probably showcasing some areas where our internal vs external apis are weird.

like, everything internal is requires things to be singleton/global, but the API still allows for instances.

Reply:
Yeah, to clarify, the method would be a class method `Tracer`

Reply:
> Yeah so one of the reasons that traces are module-level structure is that a trace is not related or dependent on a particular Tracer instance.

That's a good point.
If that's really true and you don't care about the `Tracer` object, then I think this design is totally fine.

> Also, using two Tracer instances at the same time will not make any intuitive sense since context management is global/execution specific so multiple tracer behaviour should be documented.

Right, though you could have a context management per-tracer, right? I know the context is a global variable for now, but why wouldn't be attached to a `Tracer`?